onnxruntime/contrib_ops/cpu/quantization/matmul_integer_to_float.cc

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "matmul_integer_to_float.h"
+
+#include "core/mlas/inc/mlas.h"
+#include "core/providers/common.h"
+#include "core/providers/cpu/math/matmul_helper.h"
+#include "core/util/math_cpuonly.h"
+#include "core/util/qmath.h"
+
+namespace onnxruntime {
+namespace contrib {
+
+#define REGISTER_MATMUL_INTEGER_TO_FLOAT(T)                             \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(                                        \
+      MatMulIntegerToFloat,                                             \
+      kMSDomain,                                                        \
+      1,                                                                \
+      T,                                                                \
+      kCpuExecutionProvider,                                            \
+      KernelDefBuilder()                                                \
+          .TypeConstraint("T1", DataTypeImpl::GetTensorType<uint8_t>()) \
+          .TypeConstraint("T2", DataTypeImpl::GetTensorType<T>())       \
+          .TypeConstraint("T3", DataTypeImpl::GetTensorType<float>()),  \
+      MatMulIntegerToFloat<uint8_t, T>);
+
+REGISTER_MATMUL_INTEGER_TO_FLOAT(int8_t)
+REGISTER_MATMUL_INTEGER_TO_FLOAT(uint8_t)
+
+template <typename T1, typename T2>
+Status MatMulIntegerToFloat<T1, T2>::Compute(OpKernelContext* ctx) const {
+  const Tensor* a = ctx->Input<Tensor>(0);
+  const Tensor* b = ctx->Input<Tensor>(1);
+  ORT_ENFORCE(a != nullptr && b != nullptr);
+
```

Comment:
nit: If the schema says these are required inputs the onnx checker should have validated that already